core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/Dependency.java

```diff
@@ -220,4 +222,17 @@ public static boolean isEqual(List<Dependency> dependencies, Map<String, Resourc
 		}
 	}
 
+	/**
+	 * Searches the given dependency array for the first dependency that has the matching scope.
+	 * 
+	 * @param scope - the scope to look for
+	 * @return an {@link Optional} value of a {@link Dependency} entry.
+	 */
+	public static Optional<Dependency> find(List<Dependency> dependencies, String scope) {
```

Comment:
The asymmetry makes me feel a bit worried. `TerminologyResource#getDependency(String)` has an implicit fallback lookup using the same settings key as the scope (if present), whereas if someone gets their hands on a dependency list and uses this method, they get no help in retrieving the value from elsewhere. Could this be moved entirely to `TerminologyResource`?

Reply:
The settings counterparts will be removed in an upcoming PR as those were only valid for pre-9.x data.